recipes-openxt/xenclient-tpm-scripts/xenclient-tpm-scripts/pcr1-detect.sh

```diff
@@ -25,5 +25,18 @@ if [ "${VENDOR_ID}" = "8086" ]; then
             MANUFACTURER=$(dmidecode --type 1 | sed -n -re 's/.*Manufacturer: (.*)$/\1/p' | cut -d ' ' -f 1)
             VERSION=$(dmidecode --type 1 | sed -n -re 's/.*Version: (.*)$/\1/p')
             ;;
+        5910)
+            MANUFACTURER=$(dmidecode --type 1 | \
+                           sed -n -re 's/.*Manufacturer: (.*)$/\1/p' | \
+                           cut -d ' ' -f 1)
+            PRODUCT=$(dmidecode --type 1 | \
+                      sed -n -re 's/.*Product Name: (.*)$/\1/p')
+            if [ "$MANUFACTURER" = "Dell" ] && \
+               [ "$PRODUCT" = "Latitude 5580" ] ; then
```

Comment:
I foresee this growing to a large list of quirks over time. Would it make more sense to make an install/answerfile option to include or exclude PCR[1]? This should be accompanied with some sort of disclaimer citing what the security implications of ignoring PCR[1] are to the overall trust of the system.

Reply:
I was going for "just works" to avoid adding knobs.  But like you say, it does change the security.  If a user doesn't own a dock, then they could leave it enabled.  But then later they purchase one and it throws off the measurement.

Regarding the list size, yes, it could grow quite large.  I would prefer to base detection off some attribute rather than the manufacturer and product.  I just don't know what attribute that would be.  USB-C port?  That could be too broad, and I'm not sure how to check it.

You were right on the telecon.  It's the USB NIC that was changing the measurement and not an Option ROM.  I updated the patch description to reflect reality.

Reply:
Yeah, I've been wanting to add that quirk for some time myself, after seeing machines that measure stupid stuff into PCR1.
The reason I've never done it is that if you don't measure PCR1, you can't really guarantee anymore that the hardware/firmware haven't been tampered with, which pretty much means all bets are off... Unless you know exactly what's measured into PCR1, and it happens to not be anything important.
That being said, it's useful to be able to exclude PCR1, but I agree with @pearsonk that the user should be made aware of it, or even have the ability to decide whether or not to include it.

Reply:
Ah, the USB/PCR[1] issue is fairly ubiquitous in KBL+ systems. This can be seen by adding/removing drives or even changing where a keyboard/mouse are plugged in on a system. It is very annoying and we are pushing for it to get changed.